src/player.cpp

```diff
@@ -7857,31 +7857,26 @@ void player::suffer()
         g->m.add_field(posx, posy, fd_web, 1); //this adds density to if its not already there.
     }
 
-    if (has_trait("RADIOGENIC") && int(calendar::turn) % 50 == 0 && radiation > 0) {
-        if (radiation > 10) {
-            radiation -= 10;
-            healall(1);
-        } else {
-            if(x_in_y(radiation, 10)) {
-                healall(1);
-            }
-            radiation = 0;
+    if( has_trait("RADIOGENIC") && int(calendar::turn) % MINUTES(10) == 0 && radiation > 0 ) {
+        // At 100 irradiation, 6 times faster than REGEN
+        if( x_in_y( radiation, 100 ) ) {
+            healall( 1 );
```

Comment:
Confounding factor: this checks three times as often as Regeneration (10 min cycle, rather than the 30-min effect-clock), so markedly more effective.  I'm thinking move the check to 30 min.  Might consume 5 radiation in the process.